web_relational_enhancer/static/src/js/field_enhancer.js

```diff
@@ -0,0 +1,309 @@
+odoo.define('web_relational_enhancer.field_enhancer', function(require) {
+    /**
+       Enhance many2one dropdown and many2many_tags tag
+       with additional fields on the 'display_name'
+     */
+
+    var relational = require('web.form_relational'),
+        common     = require('web.form_common'),
+        core       = require('web.core'),
+        Model      = require('web.DataModel'),
+        _t         = core._t,
+        data       = require('web.data')
+
+    function getRelationalField(field, entryList, fields, shouldSkipEntries=true) {
+        let dataset = new data.DataSet(this, field.relation),
+            def     = new $.Deferred(),
+            ids     = _.pluck(_.filter(entryList, entry => {
+                        return ( !shouldSkipEntries || !(entry.label.includes('Create') || entry.label.includes('Search')))
+            }), 'id')
+
+        try {
+            dataset.read_ids(ids, fields).then(res => {
+                def.resolve(res)
+            })
+        } catch(err) {
+            console.log(err)
+            def.reject()
+        }
+
+        return def
+    }
+
+    function enhanceMany2ManyTag(data) {
+        /**
+           If 'display_fields' (comma-separated fields names) is present in 'options'
+           this method adds them in the displayed tag.
+        */
+        let self = this
+
+        if (data.length && 'display_fields' in this.options) {
+            let fields  = this.options['display_fields'].split(','),
+                results = getRelationalField(this.field, data, fields, false),
+                def     = new $.Deferred()
+
+            results.done(res => {
+                _(res).each((obj, i) => {
+                    _(fields).each(field => {
+                        if ( obj[field] ) {
+                            data[i].display_name += `\xa0-\xa0${obj[field]}`
+                        } else {
+                            console.log(`Field: '${field}' does not exist in model: ${self.field.relation}`)
+                        }
+                    })
+                })
+
+
+                def.resolve(data)
+            }).fail(err => console.log(`Error getting relational fields. Error: ${err}`))
+
+            return def
+        } else
+            return data
+    }
+
+    function enhanceMany2OneEntryLabel(result) {
```

Comment:
Can you move the functions to the field? This will enable other developers to change the behaviors in other modules.